src/data/actions/programEnrollments.js

```diff
@@ -0,0 +1,59 @@
+import {
+  PROGRAMS_REQUEST,
+  PROGRAMS_SUCCESS,
+  PROGRAMS_FAILURE,
+  CLEAR_PROGRAMS,
+} from '../constants/programEnrollments';
+import LmsApiService from '../services/LmsApiService';
+
+const fetchProgramEnrollmentOverviewRequest = () => ({
+  type: PROGRAMS_REQUEST,
+});
+
+const fetchProgramEnrollmentOverviewSuccess = data => ({
+  type: PROGRAMS_SUCCESS,
+  payload: {
+    data,
+  },
+});
+
+const fetchProgramEnrollmentOverviewFailure = error => ({
+  type: PROGRAMS_FAILURE,
+  payload: {
+    error,
+  },
+});
+
+
+const clearProgramEnrollmentOverviewEvent = () => ({ type: CLEAR_PROGRAMS });
+
+const fetchProgramEnrollmentOverview = options => (
+  (dispatch) => {
+    dispatch(fetchProgramEnrollmentOverviewRequest(options));
+    return LmsApiService.fetchProgramEnrollmentOverview(options)
+      .then((response) => {
+        dispatch(fetchProgramEnrollmentOverviewSuccess(response.data));
+      })
+      .catch((error) => {
+        // This endpoint returns a 404 if no data exists,
+        // so we convert it to an empty response here.
+        if (error.response.status === 404) {
+          dispatch(fetchProgramEnrollmentOverviewSuccess({ results: [] }));
```

Comment:
is this the right behavior for a 404? do we detect an empty result set further downstream and render a not found page or just display a page with empty results?

Reply:
Good question, I am not quite sure too. I then implemented the way edx-portal does. @adamstankiewicz  Do you have any thoughts on this?